### PR TITLE
feat(graph): create a local hardhat + graph-node docker compose

### LINKED ISSUE #234: chore: add in ability to run subgraph locally
## Description 

The goal here is to setup subgraph locally thus making it easier to run a local node and test your queries without having to actual deploy to a hosted service maintained by the graph

### PR DESCRIPTION
**Pull request by vacekj**
@thelostone-mc @nfrgosselin I need some additional clarification on #234. What I added in this PR is a docker compose, than when run, will spin up a local hardhat node, which you can deploy contracts against, then spin up a graph-node against anvil. Then you should be able to deploy contracts as usual, interact with them and the subgraph should pick everything up. I haven't yet set-up the Anvil node as a hardhat network, since I first want to get confirmation that this is the right way to go about this.

EDIT: replace anvil node with hardhat node

##### Refers/Fixes

fixes #234

### CODE REVIEW

**Comment by thelostone-mc** [comment]
Questions: 

- can we not use the hardhat node as opposed to relying on anvil ?
- additionally to deploy the contracts on local chain -> I'm assuming we'd need to write up a script for that as well right ?

**Comment by melissa-neira** [comment]
Defer to engineering 🕺

**Comment by thelostone-mc** [comment]
Could we add in documentation on the graph/README on how to run this ?

**Review by nfrgosselin — COMMENTED** [review]
Defer to engineering review

**Comment by hmrtn** [comment]
Docker compose looks good and runs just fine with frequent block updates. 

It appears to be using hardhat rather than anvil - is this correct?

**Review by hmrtn — APPROVED** [review]

**Review by nfrgosselin — APPROVED** [review]
defer to engineering approval

File: packages/graph/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        "postgres",
+        "-cshared_preload_libraries=pg_stat_statements"
+      ]
+    environment:
+      POSTGRES_USER: graph-node
+      POSTGRES_PASSWORD: let-me-in
+      POSTGRES_DB: graph-node
+      # FIXME: remove this env. var. which we shouldn't need. Introduced by
+      # <https://github.com/graphprotocol/graph-node/pull/3511>, maybe as a
+      # workaround for https://github.com/docker/for-mac/issues/6270?
```
Comment (thelostone-mc) [inline packages/graph/docker-compose.yml:54]: ^ Is this needed ?

**Review by thelostone-mc — CHANGES_REQUESTED** [review]
could we update the documentation to state how to run this  + update the PR message to state we are using hardhat node and not anvil

File: packages/graph/docker-compose.yml (hunk 1 above)
Comment (vacekj) [reply packages/graph/docker-compose.yml]: Removed

**Comment by vacekj** [comment]
> Docker compose looks good and runs just fine with frequent block updates.
> 
> It appears to be using hardhat rather than anvil - is this correct?

That is correct, updated PR text to reflect this change.

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**